src/cli.c

```diff
@@ -28,153 +28,228 @@
+    if (opt->key)
+        fprintf(stderr, "tjs: -%c requires an argument\n", opt->key);
+    else
+        fprintf(stderr, "tjs: --%s requires an argument\n", opt->name);
+}
+
+static void report_unknown_option(CLIOption *opt) {
+    if (opt->key)
+        fprintf(stderr, "tjs: unknown option -%c\n", opt->key);
+    else
+        fprintf(stderr, "tjs: unknown option --%s\n", opt->name);
+}
+
+static unsigned get_option_length(const char *arg) {
+    const char *val_start = strchr(arg, OPT_ASSIGN);
+    if (!val_start)
+        val_start = arg + strlen(arg);
+    return val_start - arg - 1;
+}
+
+static int get_option(char **arg, CLIOption *opt) {
+    /* a single `-` is not an option, it also stops argument scanning */
+    if (!**arg)
+        return false;
+    opt->length = get_option_length(*arg);
+    if (**arg == OPT_PREFIX) {
+        opt->name = *arg + 1;
+        /* `--` stops argument scanning */
+        if (!*opt->name)
+            return false;
+        *arg += opt->length + 1;
+    } else if (**arg) {
+        opt->key = **arg;
+        *arg += 1;
+    }
+    if (**arg == OPT_ASSIGN)
+        *arg += 1;
+    return true;
+}
+
+static char *get_option_value(char *arg, int argc, char **argv, int *optind) {
+    if (*arg)
+        return arg;
+    if (*optind >= argc)
+        return NULL;
+    char *value = argv[*optind];
+    if (*value == OPT_PREFIX)
+        return NULL;
+    *optind += 1;
+    return value;
 }
 
 int main(int argc, char **argv) {
-    TJSRuntime *qrt;
-    JSContext *ctx;
-    int optind;
-    char *expr = NULL;
-    char *override_filename = NULL;
-    int interactive = 0;
-    int empty_run = 0;
-    int strict_module_detection = 0;
+    TJSRuntime *qrt = NULL;
+    JSContext *ctx = NULL;
+    int exit_code = EXIT_SUCCESS;
+
+    Flags flags = { .interactive = false,
+                    .empty_run = false,
+                    .strict_module_detection = false,
+                    .eval_expr = NULL,
+                    .override_filename = NULL };
 
     TJS_SetupArgs(argc, argv);
 
-    /* cannot use getopt because we want to pass the command line to
-       the script */
-    optind = 1;
-    while (optind < argc && *argv[optind] == '-') {
+    /* cannot use getopt because we want to pass the command line to the script */
+    int optind = 1;
+    while (optind < argc && *argv[optind] == OPT_PREFIX) {
         char *arg = argv[optind] + 1;
-        const char *longopt = "";
-        /* a single - is not an option, it also stops argument scanning */
-        if (!*arg)
+        CLIOption opt = { .key = 0, .name = NULL, .length = 0 };
+        if (!get_option(&arg, &opt))
             break;
-        optind++;
-        if (*arg == '-') {
-            longopt = arg + 1;
-            arg += strlen(arg);
-            /* -- stops argument scanning */
-            if (!*longopt)
-                break;
+        optind += 1;
+        /* combining short options is NOT supported */
```

Comment:
why not? it could make sense to use the prefload option in conjuction with the eval expr option.

Reply:
Combining in sense of joining them together: `-h -v` becomes `-hv`. (Given example does not have sense but you'll get the idea.) Why? Because it is damn hard to implement it and it even node's cli doesn't support it so I didn't waste time trying...

Reply:
Ah, gotcha. Ignore my comment then.